Cleaned up AdminUserDetails, Service, and AdminProvisioningService

nathandmoore opened the pull request:
Affects BroadleafCommerce/QA#3113

nathandmoore committed:
Tied whether the promotion sandbox is enabled to a user permission

Now, you can have individual users skip the promotion step rather than choosing only between all or none.
You can still toggle this behavior globally.

ROLE_SKIP_APPROVAL

Addresses BroadleafCommerce/QA#3113

phillipuniverse commented:
This seems a bit strange to muddy the waters here, where there is no knowledge of "approvals" at all, instead of just in BroadleafCommerce/Enterprise#982. Why does this code need to be here in core and not only in Enterprise?

nathandmoore commented:
I saw that Sandbox and Multitenant copy domains are already in common for example and just figured that if it was like that already, I could avoid adding multiple enterprise extensions. But I'll move it since that's not really a strong argument.

nathandmoore committed:
Moved skip approval logic to enterprise and added to EnterpriseAdminUserProvisioningService to cover LDAP and other external user authentications methods

Left refactoring to core classes to make it easier to extend particular pieces of the logic as desired

Addresses BroadleafCommerce/QA#3113

nathandmoore committed:
Fixed serialVersionUID

Addresses BroadleafCommerce/QA#3113
